src/OrleansRuntime/Silo/Silo.cs

```diff
@@ -137,15 +138,29 @@ public string ClusterId
         /// Test hook connection for white-box testing of silo.
         /// </summary>
         public TestHooks TestHook;
+
+        /// <summary>
+        /// Just for unit tests :(
+        /// </summary>
+        /// <param name="name"></param>
+        /// <param name="siloType"></param>
+        /// <param name="config"></param>
+        public Silo(string name, SiloType siloType, ClusterConfiguration config)
+            : this(name, siloType, null, config, null)
+        {
+            var services = new ServiceCollection();
+            StartupBuilder.RegisterSystemTypes(services);
+            Services = services.BuildServiceProvider();
```

Comment:
This might be out of the scope of this PR. but speaking of dependency injection, ideally, Silo class shouldn't be responsible for registerSystemType(services). it would be nice that we decouple registerSystemType with Silo class. 
In this way,
1. we reduce responsibility from Silo class, make it more maintainable. Currently, silo class is responsible for a lot of things, many of those are not necessarily be Silo class's responsibility. 
2. internal system services will be accessible from outside Silo class, we have an issue https://github.com/dotnet/orleans/issues/934 which wants to make system services accessible from application layer. By moving RegisterSystemTypes outside of Silo class, this could potentially be the first step to resolve issue 934.